README.md

```diff
@@ -68,35 +68,33 @@ Override these rules in `tslint.json`:
 ## Inline Linting in Text Editors
 
 
-`tslint-config-shopify` comes with a few rules that require type-checking to be enabled, however `tslint` uses a [private API](https://github.com/palantir/tslint/issues/1445#issuecomment-236330061) to resolves rules that require `type-checking`.
-
-For this reason, text editors will fail when trying to resolve rules that require `type-checking`.
-
-To resolve this issue, you can extend the untyped configuration
+TSLint comes with a few rules that rely on a [private TypeScript API](https://github.com/palantir/tslint/issues/1445#issuecomment-236330061) in order to work. Because of this, text editors will fail when trying to resolve these rules. None of these rules are included in the default configurations so that you can continue to get linter feedback in your editor, but these rules are still desireable for the "full" linting check. To accomplish this, include the base configuration(s) in your project’s root `tslint.json` file:
 
 ```json
+// In ./tslint.json
 {
-  "extends": ["tslint-config-shopify/untyped"]
+  "extends": ["tslint-config-shopify"]
 }
 ```
 
-However, this will not enforce rules that require type-checking.
+Then, in a separate configuration file, extend the `typed` variant, in addition to your base configuration:
 
-To enable the `type-checking` rules to be validated in your script, it is [recommended](https://github.com/Microsoft/vscode-tslint/issues/70#issuecomment-263120859) that you run your CLI `tslint` script against the base configuration.
-
-Eg: `./config/tslint-full.json`
 ```json
+// In ./config/tslint/full.json
 {
-  "extends": ["tslint-config-shopify"]
+  "extends": [
+    "../../tslint.json",
+    "tslint-config-shopify/typed"
+  ]
 }
 ```
 
-Then update your CLI tslint script to run against `tslint-full.json`.
+And, in your `lint` script, make sure to run it with the appropriate command-line arguments to get the full benefit of this configuration:
 
 ```json
 {
   "scripts": {
-    "tslint": "tslint -c ./config/tslint-full.json './src/**/*.{ts,tsx}' --project tsconfig.json --type-check"
+    "tslint": "tslint --config ./config/tslint/full.json './src/**/*.{ts,tsx}' --project tsconfig.json --type-check"
```

Comment:
I also noticed a 1s increase in performance within this repo when we ran `tslint`. If we update subsequent repo's we should remove `--type-check`, I think we'll see some a small performance boost.